src/parser.ts

```diff
@@ -27,28 +27,29 @@ export function parse(
 
     const lines: string[] = text.replaceAll("\r\n", "\n").split("\n");
     for (let i = 0; i < lines.length; i++) {
-        if (lines[i].length === 0) {
+        const currentLine = lines[i];
+        if (currentLine.length === 0) {
             if (cardType) {
                 cards.push([cardType, cardText, lineNo]);
                 cardType = null;
             }
 
             cardText = "";
             continue;
-        } else if (lines[i].startsWith("<!--") && !lines[i].startsWith("<!--SR:")) {
-            while (i + 1 < lines.length && !lines[i].includes("-->")) i++;
+        } else if (currentLine.startsWith("<!--") && !currentLine.startsWith("<!--SR:")) {
+            while (i + 1 < lines.length && !currentLine.includes("-->")) i++;
             i++;
             continue;
         }
 
         if (cardText.length > 0) {
             cardText += "\n";
         }
-        cardText += lines[i];
+        cardText += currentLine.trim();
```

Comment:
For the card text, we should probably do a `trimEnd()` to be on the safer side. For instance, if my card contains code that's indented we'll lose the indentation if we trim both ends